sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -311,45 +319,71 @@ private AnalyzeBatchActionsResult toAnalyzeTasks(AnalyzeJobState analyzeJobState
+                    toRecognizePiiEntitiesResultCollection(taskItem.getResults()));
+                TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
+                    taskItem.getLastUpdateDateTime());
+                recognizePiiEntitiesActionResults.add(actionResult);
+            }
         }
         if (!CoreUtils.isNullOrEmpty(keyPhraseExtractionTasks)) {
-            extractKeyPhrasesActionResults = IterableStream.of(keyPhraseExtractionTasks.stream()
-                .map(taskItem -> {
-                    ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
-                    ExtractKeyPhrasesActionResultPropertiesHelper.setResult(actionResult,
-                        toExtractKeyPhrasesResultCollection(taskItem.getResults()));
-                    TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
-                        taskItem.getLastUpdateDateTime());
-                    return actionResult;
-                })
-                .collect(Collectors.toList()));
+            for (int i = 0; i < keyPhraseExtractionTasks.size(); i++) {
+                final TasksStateTasksKeyPhraseExtractionTasksItem taskItem = keyPhraseExtractionTasks.get(i);
+                final ExtractKeyPhrasesActionResult actionResult = new ExtractKeyPhrasesActionResult();
+                ExtractKeyPhrasesActionResultPropertiesHelper.setResult(actionResult,
+                    toExtractKeyPhrasesResultCollection(taskItem.getResults()));
+                TextAnalyticsActionResultPropertiesHelper.setCompletedAt(actionResult,
+                    taskItem.getLastUpdateDateTime());
+                extractKeyPhrasesActionResults.add(actionResult);
+            }
+        }
+
+        final List<TextAnalyticsError> errors = analyzeJobState.getErrors();
+        if (!CoreUtils.isNullOrEmpty(errors)) {
+            for (TextAnalyticsError error : errors) {
+                final String[] targetPair = parseActionErrorTarget(error.getTarget());
+                final String taskName = targetPair[0];
+                final Integer taskIndex = Integer.valueOf(targetPair[1]);
+                final TextAnalyticsActionResult actionResult;
+                if ("entityRecognitionTasks".equals(taskName)) {
+                    actionResult = recognizeEntitiesActionResults.get(taskIndex);
+                } else if ("entityRecognitionPiiTasks".equals(taskName)) {
+                    actionResult = recognizePiiEntitiesActionResults.get(taskIndex);
+                } else if ("keyPhraseExtractionTasks".equals(taskName)) {
+                    actionResult = extractKeyPhrasesActionResults.get(taskIndex);
+                } else {
+                    throw logger.logExceptionAsError(new RuntimeException(
+                        "Invalid task name in target reference, " + taskName));
+                }
+
+                TextAnalyticsActionResultPropertiesHelper.setIsError(actionResult, true);
```

Comment:
I think this logic does not insert the errors in a sorted way with respect to the ordering of the input actions list, right?

Reply:
recognizeEntitiesActionResults.get(taskIndex)  recognizePiiEntitiesActionResults.get(taskIndex) or extractKeyPhrasesActionResults.get(taskIndex) get the object from the sorted output. So there is no need to reorder the errors.